inference-engine/ie_bridges/python/tests/test_InferRequest.py

```diff
@@ -544,9 +551,100 @@ def test_query_state(device):
-    with pytest.raises(ValueError) as e:
-        ones_arr = np.ones(shape=(1, 800), dtype=np.float32)
-        mem_state.state.buffer[:] = ones_arr
-    assert "assignment destination is read-only" in str(e.value)
-    assert mem_state.name == 'id_1'
+
+    assert mem_state.name == 'var_id_667'
     assert mem_state.state.tensor_desc.precision == 'FP32'
+
+    for i in range(1, 10):
+        res = exec_net.infer({"input_data": [1] * 10})
+        expected_res = np.array([i] * 10, dtype=float)
+        assert np.allclose(res['MemoryAdd'], expected_res, atol=1e-6), \
+            "Expected values: {} \n Actual values: {} \n".format(expected_res, res)
+
+
+@pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU",
+                    reason=f"Can't run test on device {os.environ.get('TEST_DEVICE', 'CPU')}, "
+                    "Memory layers fully supported only on CPU")
+def test_query_state_reset(device):
+    ie_core = ie.IECore()
+    if device == "CPU":
+        if ie_core.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin")
+
+    import ngraph as ng
+    from ngraph.impl import Function
+
+    input_data = ng.parameter([10], name="input_data", dtype=np.float32)
+    rv = ng.read_value(input_data, "var_id_667")
+    add = ng.add(rv, input_data, name="MemoryAdd")
+    node = ng.assign(add, "var_id_667")
+    res = ng.result(add, "res")
+    func = Function(results=[res], sinks=[node], parameters=[input_data], name="name")
+    caps = Function.to_capsule(func)
+
+    net = ie.IENetwork(caps)
+    ie_core = ie.IECore()
+    exec_net = ie_core.load_network(network=net, device_name=device, num_requests=1)
+    request = exec_net.requests[0]
+    mem_states = request.query_state()
+    mem_state = mem_states[0]
+
+    assert mem_state.name == 'var_id_667'
+    assert mem_state.state.tensor_desc.precision == 'FP32'
+
+    for i in range(1, 10):
+        # reset initial state of ReadValue to zero
+        mem_state.reset()
+        res = exec_net.infer({"input_data": [1] * 10})
+
+        # always ones
+        expected_res = np.array([1] * 10, dtype=np.float32)
+
+        assert np.allclose(res['MemoryAdd'], expected_res, atol=1e-6), \
+            "Expected values: {} \n Actual values: {} \n".format(expected_res, res)
+
+
+@pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU",
+                    reason=f"Can't run test on device {os.environ.get('TEST_DEVICE', 'CPU')}, "
+                    "Memory layers fully supported only on CPU")
+def test_query_state_write_buffer(device):
+    ie_core = ie.IECore()
+    if device == "CPU":
+        if ie_core.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin")
+
+    import ngraph as ng
+    from ngraph.impl import Function
+    from openvino.inference_engine import TensorDesc, Blob
+
+    input_data = ng.parameter([10], name="input_data", dtype=np.float32)
```

Comment:
Let's move s network creation to another function as it was duplicated in these tests.

Reply:
done